redesign/base_generator.py

```diff
@@ -0,0 +1,17 @@
+"""Contains abstract class from which generators will inherit."""
+
+import abc
+
+
+class BaseGenerator(metaclass=abc.ABCMeta):
+    """Abstract generator class."""
+
+    def __init__(self, profile, options):
+        """Initialize generator object."""
+        self.profile = profile.report()
+        self.options = options
+
+    @abc.abstractmethod
+    def synthesize(self):
+        """Make synthetic data."""
+        raise NotImplementedError()
```

Comment:
I would expect this to take in at the very least `samples` or something of the like to specify how much we want.

This would be a great case for also allowing additional args to pass through.

e.g.
```python
def synthesize(self, samples, *args, **kwargs):
    .
    .
    .
```